src/vips/Image_methods.lua

```diff
@@ -497,6 +497,16 @@ function Image_method:remove(name)
     return vips_lib.vips_image_remove(self.vimage, name) ~= 0
 end
 
+function Image_method:get_gainmap()
+    collectgarbage("stop")
```

Comment:
I'm not certain you need to stop and restart the GC -- with the API change `vips_image_get_gainmap` now returns a true reference, so your pointer should be safe.

Reply:
It should be, but there are some oddities in Lua 5.4 which cause a crash (illegal machine instruction) when running the `uhdr.lua` script (still seen in https://github.com/libvips/lua-vips/compare/ca7df971d6a3a34ce9a076e8c3e238812c1452f2..7e8f5c44813ee311e0acf1e3f5c4bd0bf97c175d) via `lua5.4 uhdr.lua`. 
It works fine with `lua5.3` and `luajit` without stopping the GC.